subprojects/distributions-dependencies/distributions-dependencies.gradle.kts

```diff
@@ -14,47 +14,49 @@
  * limitations under the License.
  */
 
-// This project at the bottom of the project hierarchy enforce the "platform" for the Gradle distribution.
-// We want the versions that are packaged in the distribution to be used everywhere (e.g. in all test scenarios)
-// Hence, we lock the versions down here for all other subprojects.
-//
-// Note:
-// We use strictly here because we do not have any better means to do this at the moment.
-// Ideally we wound be able to say "lock down all the versions of the dependencies resolved for the distribution"
+/**
+ * This project at the bottom of the project hierarchy enforce the "platform" for the Gradle distribution.
+ * We want the versions that are packaged in the distribution to be used everywhere (e.g. in all test scenarios)
+ * Hence, we lock the versions down here for all other subprojects.
+ *
+ * Note:
+ * We use strictly here because we do not have any better means to do this at the moment.
+ * Ideally we wound be able to say "lock down all the versions of the dependencies resolved for the distribution"
+ */
 plugins {
-    id "base"
+    base
 }
 dependencies {
     constraints {
-        libraries.values().each { lib ->
-            'default'(lib.coordinates) {
+        libraries.keys.forEach { libId ->
+            add("default", library(libId)) {
                 version {
-                    strictly lib.version
-                    if (lib.because) { because lib.because }
+                    strictly(libraryVersion(libId))
                 }
+                because(libraryReason(libId))
```

Comment:
:o:
![image](https://user-images.githubusercontent.com/132773/36534206-2bf4d7c2-17c6-11e8-8860-46cbc8fbce56.png)

As a follow up you should add `@Nullable` to the `DependencyConstraint.because(reason)` method.

Reply:
Good catch. (I wonder why did I not see warning.) Fixed: e14428c